src/Microsoft.Extensions.ProjectModel.Sources/DotNet/DotNetProjectContext.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.DotNet.ProjectModel;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.Frameworks;
+
+namespace Microsoft.Extensions.ProjectModel
+{
+    internal class DotNetProjectContext : IProjectContext
+    {
+        private readonly ProjectContext _project;
+        private readonly OutputPaths _paths;
+        private readonly bool _isExecutable;
+        private readonly JObject _rawProject;
+        private readonly CommonCompilerOptions _compilerOptions;
+
+        public DotNetProjectContext(ProjectContext wrappedProject, string configuration, string outputPath)
+        {
+            if (wrappedProject == null)
+            {
+                throw new ArgumentNullException(nameof(wrappedProject));
+            }
+
+            if (string.IsNullOrEmpty(configuration))
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            using (var stream = new FileStream(wrappedProject.ProjectFile.ProjectFilePath, FileMode.Open, FileAccess.Read))
+            using (var streamReader = new StreamReader(stream))
+            using (var jsonReader = new JsonTextReader(streamReader))
+            {
+                _rawProject = JObject.Load(jsonReader);
```

Comment:
question is, do we need to. i know the `FindProperty` is defined on the interface, but all the information of a `project.json` is exposed through the project model.  the `project.json` shouldn't be treated as a simple dictionary.